README.md

```diff
@@ -1,28 +1,60 @@
+ 
+NumPyro is designed to be *lightweight* and focuses on providing a flexible substrate that users can build on:
+
+ - **Pyro Primitives:** NumPyro programs can contain regular Python and NumPy code, in addition to [Pyro primitives](http://pyro.ai/examples/intro_part_i.html) like `sample` and `param`. The model code should look very similar to Pyro except for some minor differences between PyTorch and Numpy's API. See [Examples](#Examples).
+ - **Inference algorithms:** NumPyro currently supports Hamiltonian Monte Carlo, including an implementation of the No U-Turn Sampler. One of the motivations for NumPyro was to speed up Hamiltonian Monte Carlo by JIT compiling the verlet integration step that includes multiple gradient computations. With JAX, we can compose `jit` and `grad` to compile the entire integration step into an XLA optimized kernel. We also eliminate Python overhead by JIT compiling the entire tree building stage in NUTS. There is also a basic Variational Inference implementation for reparameterized distributions.
+ - **Distributions:** The `numpyro.distributions` module provides distribution classes, constraints and bijective transforms. The distribution classes wrap over samplers implemented to work with JAX's [functional pseudo-random number generator](https://github.com/google/jax#random-numbers-are-different). The design of the distributions module largely follows from [PyTorch](https://pytorch.org/docs/stable/distributions.html). A major subset of the API is implemented, and it contains most of the common distributions that exist in PyTorch. As a result, Pyro and PyTorch users can rely on the same API and batching semantics as in `torch.distributions`. In addition to distributions, `constraints` and `transforms` are very useful when operating on distribution classes with bounded support.
+ - **Effect handlers:** Like Pyro, primitives like `sample` and `param` can be interpreted with side-effects using effect-handlers from the `numpyro.handlers` module, and these can be easily extended to implement custom inference algorithms and inference utilities.
+
+
+## Installation
+
+To install NumPyro with a CPU version of JAX, you can use pip:
+
+```
+pip install numpyro
+```
+
+To use NumPyro on the GPU, you will need to first [install](https://github.com/google/jax#installation) `jax` and `jaxlib` with CUDA support.
+
+You can also install NumPyro from source:
+
+```
+git clone https://github.com/pyro-ppl/numpyro.git
+# install jax/jaxlib first for CUDA support
+pip install -e .[dev]
+```
+
+## Examples
+
+
+For some examples on specifying models and doing inference in NumPyro:
+
+ - [Bayesian Regression in Numpyro](https://github.com/pyro-ppl/numpyro/blob/master/notebooks/bayesian_regression.ipynb) - Start here to get acquainted with writing a simple model in NumPyro, MCMC inference API, effect handlers and writing custom inference utilities.
```

Comment:
I think that we might need to add a link using `nbviewer` because github is usually fail to render the notebook. WDYT?

Reply:
Good idea, will change.